### ISSUE TITLE
Also ship with DejaVu Sans Mono Italic versions

### ISSUE THREAD

**Issue by porres** [pure-data/pure-data#892]
Pd comes with DejaVu Sans Mono, but it only has normal and bold fonts and it misses italic and italic+bold fonts.

I know Pd doesn't use italic itself, but the thing is I'm working on cyclone's comment and using DejaVu Sans Mono as the default font, so it looks like Pd's comment by default, and then cyclone's comment will now be able to load italic versions of the font, that is if it exists, right now nothing happens cause we don't have it so I'm requesting this feature.

What do you think?

Thanks

**Comment by porres** [comment]
I did a simple test and just included DejaVuSansMono-BoldOblique.ttf and DejaVuSansMono-Oblique.ttf into "font" in Pd's source, compliled it and, voilá, it worked fine for macOS. Wasn't really expecting it'd be that easy, but it still seems it has to be a little bit more elaborate than that. Anyway, will try and see and then maybe even send a PR.

**Comment by umlaeute** [comment]
why can't you just ship the fonts with cyclone?

**Comment by porres** [comment]
I had thought of that and tried it, but sending this or any other font inside the cyclone folder doesn't really work for cyclone/comment (it'd be cool if it did though). Then I tried just adding _DejaVuSansMono-BoldOblique.ttf_ and _DejaVuSansMono-Oblique.ttf_ into "resources/font" in the mac application and it worked. Then also added to the source and compiled, which ended up copying the files to the same folder and also worked then.

Anyway, DVSM  is a dependency for the puredata package in Linux, and it comes with italic, why not also ship then with Pd for macOS? What would be the opposition? Once we're already shipping it (but only half of it).

**Comment by porres** [comment]
I see that in Preferences => Path it says it should also load "fonts", I guess it's cool if you could ship fonts with your patch, so that pd finds it either in the same folder where the patch is, and then via the search paths and etc

**Comment by porres** [comment]
> I see that in Preferences => Path it says it should also load "fonts", I guess it's cool if you could ship fonts with your patch, so that pd finds it either in the same folder where the patch is, and then via the search paths and etc

what I mean is that this is not actually true and not happening right now

**Comment by umlaeute** [comment]
that's probably only there because externals such as Gem search for fonts in the `path` (i don't think that it was *me* who put that in, but the history is lost).

**Comment by porres** [comment]
> externals such as Gem search for fonts in the path

does it also search for them in the patch's directory?

**Comment by umlaeute** [comment]
i don't understand how Gem's searching for fonts is related to the issue at hand¹. (I only mentioned it because Gem's behaviour is likely to be the cause for the text in the preferences menu). but yes, it tries to open the fonts in the canvas' directory first, using `open_via_path()` and passing the canvas' path as the default directory.

¹ Gem uses it's "own" font-rendering engine, that directly loads TrueTypeFont files. this is quite different from what Tcl/Tk does, so I don't think that you can borrow code for cyclone's `comment` implementation

**Comment by porres** [comment]
I'm just thinking if mentioning 'fonts' in there is pertinent, but from your answer, I think not and thus I don't think Pd should mention this as it can't do it on its own - but then changing this would be a separate issue.

**Comment by porres** [comment]
> I don't think that you can borrow code for cyclone's comment implementation

yeah, that would be another thing. I can see how nice it'd be, but I wouldn't know how to make it possible, if it is (I assume it should be)

**Comment by danomatika** [comment]
Loading fonts via a 3rd party path won't work on Windows since the font wouldn't be found by pdfontloader.

**Comment by umlaeute** [comment]
?

afaik, pdfontloader allows us to specify the font to be loaded. at least this is how Pd is using it:
https://github.com/pure-data/pure-data/blob/e2ef8158e83900734d263a3fda343a47ed0c9e3a/tcl/pd-gui.tcl#L409-L413

why would that not work with a gui-plugin? (the gui-plugin knows about its path, so it can calculate the full path to the font it tries to load via `pdfontloader::load`

**Comment by danomatika** [comment]
My meaning was: "If you expect the Tcl/Tk GUI to automatically find a font in your 3rd party path, load it, and make it available. You are wrong." It's currently only used to load form Pd's root "font/" directory and I was never thinking about people installing custom fonts there, but I suppose they *could*.

Of course, if you *manually* do all of this but then again you would need to know about pdfontloader for windows to begin with...
